xarray/core/computation.py

```diff
@@ -1482,6 +1506,53 @@ def _cov_corr(
         return corr
 
 
+def _weighted_cov_corr(
```

Comment:
I wonder if we really need two codepaths for the weighted and unweighted options? Perhaps we can merge them into one?

Reply:
I agree, but was hesitant to modify the original function. An option to merge the two funcitons together is to use weights=1 by default (i.e. when no weights have been specified), but this would probably slow down unnecessarily the unweighted computation?
Another option is to have the .weighted calls inside if blocks. This would be in three places. In both cases it would be good to check that the function produces the same results than before. Where can I see if the cov and corr functions have some tests that I can reuse for that purpose?

Reply:
I think it's good! I can see where your original hesitation, but I do think it's still a better option to have some slightly verbose code in the function rather than two almost duplicative functions.